driver-core/src/main/java/com/datastax/driver/core/Requests.java

```diff
@@ -591,7 +591,12 @@ public void encode(Prepare msg, ByteBuf dest, ProtocolVersion version) {
 
             @Override
             public int encodedSize(Prepare msg, ProtocolVersion version) {
-                return CBUtil.sizeOfLongString(msg.query);
+                int size = CBUtil.sizeOfLongString(msg.query);
+
+                if (version.compareTo(ProtocolVersion.V5) >= 0) {
+                    size += 4; //flags
+                }
```

Comment:
It would be better to use `PREPARED_METADATA_CHANGES.isSupportedBy(version)` for this test.

Reply:
was following the logic in `encode` which checks the protocol version, should we change that to?

Reply:
Mmm actually the flag does not relate directly to this feature, but instead to per-query keyspaces, which are not supported at all in the driver at this point. Keep the version test, my bad.